jerry-core/vm/vm.c

```diff
@@ -1569,6 +1579,13 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
           }
 
           const int index = (int) (opcode_data >> VM_OC_NON_STATIC_SHIFT) - 2;
+
+          if (JERRY_UNLIKELY (ecma_compare_ecma_string_to_magic_id (prop_name_p, LIT_MAGIC_STRING__PROTO__)))
+          {
+            result = ecma_builtin_object_object_set_prototype_of (stack_top_p[index], left_value, true);
```

Comment:
What if instead of adding an argument to the `set_prototype_of` function, we call it as we would normally and release the resulting value, and just set the vm result to undefined? Would that be feasible?
@rerobika What is your opinion?

Reply:
+1

Reply:
I've made a new method based on ECMA-262 v6, B.2.2.1.2 to separate the different behaviours, so if we want to set the __proto__ accessor poperty, we use this new method.